services/userlog/pkg/service/filter.go

```diff
@@ -0,0 +1,102 @@
+package service
+
+import (
+	"context"
+	"errors"
+	user "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	"github.com/cs3org/reva/v2/pkg/events"
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	"github.com/owncloud/ocis/v2/ocis-pkg/middleware"
+	settingssvc "github.com/owncloud/ocis/v2/protogen/gen/ocis/services/settings/v0"
+	"github.com/owncloud/ocis/v2/services/settings/pkg/store/defaults"
+	micrometadata "go-micro.dev/v4/metadata"
+)
+
+type UserlogFilter struct {
+	log         log.Logger
+	valueClient settingssvc.ValueService
+}
+
+func NewUserlogFilter(l log.Logger, vc settingssvc.ValueService) *UserlogFilter {
+	return &UserlogFilter{log: l, valueClient: vc}
+}
+
+// Filter removes users who should not receive an in-app notifications for the event
+func (ulf UserlogFilter) Filter(ctx context.Context, event events.Event, executant *user.UserId, users []string) []string {
```

Comment:
This method would be called as `filter.Filter`. That stutters. Maybe rename it to `Execute`? So it'd be called like `filter.Execute`.